File: LeetCode/0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int flowerbed_size = flowerbed.size();
        for (int i = 0; i < flowerbed_size; i++) {
            if (flowerbed[i])
                continue;
            int left = i - 1, right = i + 1;
            if (left < 0)
                ++left;
            if (right >= flowerbed_size)
                --right;
            if (flowerbed[left] == 0 && flowerbed[right] == 0) {
                --n;
                flowerbed[i] = 1;
            }
            if (n <= 0)
                return true;
        }
        if (n <= 0)
            return true;
        return false;
    }
};

File: LeetCode/0605-can-place-flowers/0605-can-place-flowers.ts
function canPlaceFlowers(flowerbed: number[], n: number): boolean {
    const length = flowerbed.length;

    for (let i = 0; i < length; i++) {
        if (!flowerbed[i] &&
            (i + 1 >= length || !flowerbed[i + 1]) &&
            (i - 1 < 0 || !flowerbed[i - 1])) {
            flowerbed[i] = 1;
            n--;
        }
        if (n <= 0) return true
    }
    return false
};